templates/terraform/examples/pubsub_subscription_push_with_service_account.tf.erb

```diff
@@ -0,0 +1,67 @@
+data "google_project" "current" {}
+
+# ------------------
+# CloudRun
+# ------------------
+resource "google_cloud_run_service" "<%= ctx[:primary_resource_id] %>" {
+  name     = "<%= ctx[:vars]['cloudrun_name'] %>"
+  location = "<%= ctx[:vars]['cloudrun_region'] %>"
+  provider = "google-beta"
+
+  metadata {
+    namespace = "${google_project.current.name}"
+  }
+
+  spec {
+    containers {
+      image = "gcr.io/cloudrun/hello"
+    }
+  }
+}
+
+# -------------
+# Cloud Pub/Sub
+# -------------
+resource "google_pubsub_topic" "<%= ctx[:primary_resource_id] %>" {
+  name = "<%= ctx[:vars]['topic_name'] %>"
+}
+
+resource "google_project_iam_member" "pubsub_sa_token_creator" {
+  role   = "roles/iam.serviceAccountTokenCreator"
+  member = "serviceAccount:service-${google_project.current.number}@gcp-sa-pubsub.iam.gserviceaccount.com"
+}
+
+resource "google_service_account" "<%= ctx[:primary_resource_id] %>" {
+  account_id = "<%= ctx[:vars]['service_account_id'] %>"
+}
+
+# TODO: grant role only to the CloudRun service rather than project wide.
+# (CloudRun service iam policy is not implemented on the Terraform yet)
+# ```
+# gcloud beta run services add-iam-policy-binding "${google_cloud_run_service.<%= ctx[:primary_resource_id] %>.name}" \
+#    --member="serviceAccount:${google_service_account.<%= ctx[:primary_resource_id] %>.email}" \
+#    --role=roles/run.invoker
+# ```
+resource "google_project_iam_member" "<%= ctx[:primary_resource_id] %>_run_invoker" {
+  role   = "roles/run.invoker"
+  member = "serviceAccount:${google_service_account.<%= ctx[:primary_resource_id] %>.email}"
+}
+
+resource "google_pubsub_subscription" "<%= ctx[:primary_resource_id] %>" {
+  name  = "<%= ctx[:vars]['subscription_name'] %>"
+  topic = "${google_pubsub_topic.<%= ctx[:primary_resource_id] %>.name}"
+
+  push_config {
+    # TODO: this must be CloudRun's url, which is not currently exposed on the Terraform
+    # push_endpoint = "${google_cloud_run_service.<%= ctx[:primary_resource_id] %>.status.url}"
```

Comment:
I believe this pull requests will fix this: https://github.com/GoogleCloudPlatform/magic-modules/pull/2185